odoo/tools/html_sanitize.py

```diff
@@ -0,0 +1,287 @@
+}
+
+# Extend the allowed attribute list when sanitize_attributes is set to False
+# All data-XXX attributes are already allowed, see check_attribute
+extended_safe_attrs = {
+    'aria-expanded',
+    'loading',
+    'style',
+}
+
+safe_tags = {
+    '?', 'a', 'abbr', 'acronym', 'address', 'applet', 'area', 'article', 'aside',
+    'audio', 'b', 'basefont', 'bdi', 'bdo', 'big', 'blink', 'blockquote', 'bodyb',
+    'br', 'button', 'c', 'canvas', 'caption', 'center', 'cite', 'code', 'col',
+    'colgroup', 'command', 'd', 'datalist', 'dd', 'del', 'details', 'dfn', 'dir',
+    'div', 'dl', 'dt', 'e', 'em', 'f', 'fieldset', 'figcaption', 'figure', 'font',
+    'footer', 'form', 'frameset', 'h', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'header',
+    'hgroup', 'hr', 'html', 'i', 'img', 'input', 'ins', 'isindex', 'kbd', 'keygen',
+    'l', 'label', 'legend', 'li', 'm', 'main', 'map', 'mark', 'marquee', 'math',
+    'menu', 'meter', 'n', 'nav', 'o', 'ol', 'optgroup', 'option', 'output', 'p',
+    'param', 'pre', 'progress', 'q', 'role', 'rp', 'rt', 'ruby', 's', 'samp', 'section',
+    'select', 'small', 'source', 'span', 'strike', 'strong', 'sub', 'summary', 'sup',
+    'svg', 't', 'table', 'tbody', 'td', 'textarea', 'tfoot', 'th', 'thead', 'time',
+    'tr', 'track', 'tt', 'u', 'ul', 'var', 'video', 'wbr',
+}
+
+
+# Those tags are removed and their children as well
+# If sanitize_tags is False, this list is ignored
+kill_tags = {
+    'base', 'embed', 'frame', 'head', 'iframe', 'link', 'meta',
+    'noscript', 'object', 'script', 'style', 'title',
+}
+
+# Those tag are remove even if sanitize_tags is False
+clean_tags = {
+    'meta',
+}
+
+style_safe_attrs = {
+    'background-color', 'color', 'display', 'float', 'font-family', 'font-size',
+    'font-weight', 'letter-spacing', 'line-height', 'margin', 'margin-bottom',
+    'margin-left', 'margin-right', 'margin-top', 'opacity', 'padding', 'padding-bottom',
+    'padding-left', 'padding-right', 'padding-top', 'text-align', 'text-decoration',
+    'text-transform', 'vertical-align', 'white-space'
+    # box model
+    'border', 'border-bottom', 'border-color', 'border-radius', 'border-style',
+    'border-top', 'border-width', 'height', 'max-width', 'min-height', 'min-width',
+    'width',
+    # tables
+    'border-collapse', 'border-spacing', 'caption-side',
+    'empty-cells', 'table-layout',
+    # border style
+    *{
+        f'border-{position}-{attribute}'
+        for position in ('top', 'bottom', 'left', 'right')
+        for attribute in ('style', 'color', 'width', 'left-radius', 'right-radius')
+    },
+}
+
+
+KILLED_TAG_NAME = '__<killed>__'
+re_tag_start = rf'<\s*{KILLED_TAG_NAME}[^>]*>'
+re_tag_end = rf'<\/\s*{KILLED_TAG_NAME}\s*>'
+
+re_killed_tag = re.compile(re_tag_start + '.*?' + re_tag_end, re.DOTALL)
+re_killed_empty = re.compile(re_tag_start, re.DOTALL)
+
+ATTRIBUTE_REGEX = re.compile(rf'^[{string.ascii_letters + string.digits}\-_]*$')
```

Comment:
Should we go with "DATA_ATTRIBUTE_REGEX" then since it's only used for data attributes?

Reply:
ok for me :)